mailbox/cassandra/src/main/java/org/apache/james/mailbox/cassandra/mail/CassandraMailboxMapper.java

```diff
@@ -193,4 +197,37 @@ private MailboxPath path(Mailbox mailbox) {
                 .where(eq(MAILBOX_BASE, typesProvider.getDefinedUserType(CassandraMailboxTable.MAILBOX_BASE).newValue().setString(MailboxBase.NAMESPACE, namespace).setString(MailboxBase.USER, user)))));
     }
 
+    @Override
+    public List<MailboxAnnotation> getAnnotations(Mailbox mailbox) throws MailboxException {
+        CassandraId mailboxId = (CassandraId) mailbox.getMailboxId();
+        return new CassandraAnnotationMapper(mailboxId, session).getAnnotations();
+    }
+
+    @Override
+    public List<MailboxAnnotation> getAnnotations(Mailbox mailbox, List<String> keys)
+            throws MailboxException {
+        CassandraId mailboxId = (CassandraId) mailbox.getMailboxId();
+        return new CassandraAnnotationMapper(mailboxId, session).getAnnotations()
+            .stream()
+            .filter(mailboxAnnotation -> keys.contains(mailboxAnnotation.getKey()))
+            .collect(Collectors.toList());
+    }
+
+    @Override
+    public void setAnnotation(Mailbox mailbox, List<MailboxAnnotation> mailboxAnnotations)
```

Comment:
it's not a set as it doesn't overwrite mailbox current annotations, it only do an update with given annotations